Change C++ standard to C++17 to avoid build failures due to
ICU dependency.

/data/EasyBuild-develop/software/ICU/75.1-GCCcore-13.3.0/include/unicode/localpointer.h:561:26:
error: ‘auto’ parameter not permitted in this context
  561 | template <typename Type, auto closeFunction>
      |                          

Do not use C++20, as char16_t in std::cout is not allowed starting with C++20,
causing building the tests to fail.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33bc40f..133468f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ cmake_minimum_required(VERSION 3.12.0)
 
 # Try C++14, then fall back to C++11 and C++98.  Used for feature tests
 # for optional features.
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 # Use folders (for IDE project grouping)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
-- 
2.49.0

